### PR TITLE
[daikin] add energy usage day & week

### PR DESCRIPTION
**Pull request by gitwouter**
Signed-off-by: Wouter Denayer <wouter@denayer.com>

### CODE REVIEW

**Comment by gitwouter** [comment]
I have tried several times now to fix the DCO remark but to no avail... Could someone please give me clear instructions on which git commands to use to update the author on the relevant commit?

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.stream.Collectors;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Holds information from the get_week_power_ex call.
+ *
+ * @author Wouter Denayer <wouter@denayer.com> - Added to support for weekly & daily energy reading
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:27]: The email adress shouldn't be part of the athor tag.

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    private static final Logger logger = LoggerFactory.getLogger(EnergyInfoDayAndWeek.class);
+
+    private EnergyInfoDayAndWeek() {
+    }
+
+    public static EnergyInfoDayAndWeek parse(String response) {
+
+        EnergyInfoDayAndWeek info = new EnergyInfoDayAndWeek();
+
+        logger.debug("Parsing string: \"{}\"", response);
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:48]: Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java (hunk 1 above)
Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:27]: no problem to remove, it does seem to be very frequent that email is added (cf other source files in this same binding)

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java (hunk 2 above)
Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:48]: agree, this should be trace level probably. Will change it in EnergyInfoYear as well.

Comment (fwolter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java]: It seems to me that this output is only needed during development. Can you use the debugger if you want to inspect it?

Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java]: well.. I can't... haven't worked with the debugger. I guess it should be possible. It's only one line, so why not keep it in trace?

**Review by fwolter — CHANGES_REQUESTED** [review]
There are some checkstyle warnings left. You could take a look at `target/code-analysis/report.html`.

I strongly encourage you to make you familiar with the debugger. It saves a lot of pain. You can leave the debug message.

**Comment by fwolter** [comment]
Are you finished making all changes?

**Comment by gitwouter** [comment]
> Are you finished making all changes?

yes

**Comment by fwolter** [comment]
Did you push your changes? There are still several checkstyle warnings. Here are some of them:
grafik

There are also compiler warnings in your newly added code.

**Comment by gitwouter** [comment]
> Did you push your changes? There are still several checkstyle warnings. Here are some of them:
> grafik
> 
> There are also compiler warnings in your newly added code.

for the logger I don't know what to change
1. I believe the name is correct (all lowercase). It is flagged by a naming rule made for constants, but it is not a constant.
2. it is static because it is then used in a static function, if I make it non-static then it doesn't work.

could you please advise how to address these points? Thx.

**Comment by fwolter** [comment]
It's ok to have the logger static if you need to access it in a static way. If you make it `static final` it becomes a constant. So, its name should be all upper case.

The remaining compiler warnings apply only to code you didn't touch with this PR.

Would be awesome if you could file a follow-up PR to fix these. You should see them in eclipse but also if you execute `mvn clean install`.
```
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\DaikinWebTargets.java:[218,31] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\api\airbase\AirbaseEnums.java:[179,37] The value of the field org.openhab.binding.daikin.internal.api.airbase.AirbaseEnums.AirbaseFeature.LOGGER is not used
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\DaikinHttpClientFactoryImpl.java:[48,17] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\DaikinHttpClientFactoryImpl.java:[65,45] The constructor org.eclipse.jetty.util.ssl.SslContextFactory(boolean) is deprecated
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\DaikinHttpClientFactoryImpl.java:[67,17] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\discovery\DaikinACUnitDiscoveryService.java:[82,42] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\discovery\DaikinACUnitDiscoveryService.java:[83,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\discovery\DaikinACUnitDiscoveryService.java:[91,42] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\discovery\DaikinACUnitDiscoveryService.java:[92,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[144,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[147,17] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[148,31] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[148,38] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[150,59] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[150,72] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[150,87] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[151,31] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[171,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[178,36] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[179,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[189,47] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[189,69] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[190,82] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[190,95] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[191,30] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[218,30] Type safety: Unchecked cast from org.openhab.core.types.Command to org.openhab.core.library.types.QuantityType<javax.measure.quantity.Temperature>
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[218,30] Potential null pointer access: The method toUnit(Unit) may return null
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinBaseHandler.java:[230,22] The value of the local variable thingTypeUID is not used
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[79,42] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[82,54] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[83,32] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[87,13] Redundant null check: The variable controlInfo cannot be null at this location
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[105,33] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[106,13] Redundant null check: The variable sensorInfo cannot be null at this location
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[112,36] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[113,13] Redundant null check: The variable zoneInfo cannot be null at this location
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[122,35] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[124,9] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[129,35] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[131,9] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[144,50] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[145,56] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[150,35] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[152,9] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[166,25] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[170,60] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[175,35] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[177,9] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[181,33] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[183,27] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[187,36] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[189,19] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[194,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[211,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[217,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[228,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\handler\DaikinAirbaseUnitHandler.java:[234,13] Potential null pointer access: this expression has a '@Nullable' type
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\api\EnergyInfoDayAndWeek.java:[69,42] Potential null pointer access: The method get(Object) may return null
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\api\EnergyInfoDayAndWeek.java:[83,42] Potential null pointer access: The method get(Object) may return null
[WARNING] C:\Users\fabian.wolter\Documents\GitHub\openhab-addons\bundles\org.openhab.binding.daikin\src\main\java\org\openhab\binding\daikin\internal\api\Enums.java:[130,37] The value of the field org.openhab.binding.daikin.internal.api.Enums.HomekitMode.LOGGER is not used
```

**Comment by gitwouter** [comment]
the build here on github now fails, (error when clicking the link)... why is that?

**Comment by wborn** [comment]
The build results aren't stored indefinitely. I've triggered a new build so it will show the reason again when it fails.

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
```
Comment (wborn) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:2]: This should fix the build: 

```suggestion
 * Copyright (c) 2010-2021 Contributors to the openHAB project
```

**Comment by gitwouter** [comment]
@cpmeister the build passes and I have addressed the mandatory points. Is the PR ready for merge now?

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ *
+ */
+@NonNullByDefault
+public class EnergyInfoDayAndWeek {
+    public Optional<Double> energyHeatingToday = Optional.empty();
+    public Optional<Double> energyHeatingThisWeek = Optional.empty();
+    public Optional<Double> energyHeatingLastWeek = Optional.empty();
+    public Optional<Double> energyCoolingToday = Optional.empty();
+    public Optional<Double> energyCoolingThisWeek = Optional.empty();
+    public Optional<Double> energyCoolingLastWeek = Optional.empty();
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:37]: Are these `public` by intention?

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    private static final Logger logger = LoggerFactory.getLogger(EnergyInfoDayAndWeek.class);
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:39]: As the logger is static now, the field name should be all uppercase.

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java (hunk 4 above)
Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:37]: it is the same in the other similar classes in this binding

Comment (fwolter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:37]: If they don't need to be `public`, you could make them `private`, too.

Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:37]: it has to be public

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java [hunk 6]
```diff
@@ -49,12 +51,14 @@
...
- * @author Paul Smedley <paul@smedley.id.au> - Modifications to support Airbase Controllers
+ * @author Paul Smedley - Modifications to support Airbase Controllers
  * @author Lukas Agethen - Added support for Energy Year reading, compressor frequency and powerful mode
+ * @author Wouter Denayer - Added to support for weekly & daily energy reading
+ * 
  */
 @NonNullByDefault
 public class DaikinAcUnitHandler extends DaikinBaseHandler {
-    private final Logger logger = LoggerFactory.getLogger(DaikinAcUnitHandler.class);
+    private final Logger LOGGER = LoggerFactory.getLogger(DaikinAcUnitHandler.class);
```
Comment (fwolter) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinAcUnitHandler.java:61]: As this logger is not `static`, it must be lower case.

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java (hunk 4 above)
Comment (fwolter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfoDayAndWeek.java:37]: Fields shouldn't be accessed directly by other classes. You can generate the necessary getters with Alt-Shift-S, R in eclipse.

**Review by fwolter — CHANGES_REQUESTED** [review]

**Comment by fwolter** [comment]
@gitwouter What's the state of this PR?

**Comment by gitwouter** [comment]
> @gitwouter What's the state of this PR?

for me it is ready

**Comment by fwolter** [comment]
Can you work on this comment? https://github.com/openhab/openhab-addons/pull/9602#discussion_r588872910

**Comment by gitwouter** [comment]
Fabian, I am not going to make any more changes unless it's a bug fix. I don't have the time for the moment. Feel free to take the source and make any changes you fancy.

On 31 March 2021 at 10:32:37, Fabian Wolter ***@***.***) wrote:

Can you work on this comment? #9602 (comment)

**Pull request merged**